tests/integration/integration-test.py

```diff
@@ -62,6 +64,191 @@ def test_get_notes_path(self, endpoint='/notes'):
+            for resource in response:
+                attributes = resource['attributes']
+                current_element = attributes['context']['contextType']
+                self.assertGreaterEqual(current_element, greater_element)
+                greater_element = current_element
+        else:
+            self.assertEqual(response, [])
+
+    def query_source(self, student_id, sources, endpoint='/notes'):
+        for source in sources:
+            params = {'studentId': student_id, 'sources': source}
+            response = self.get_response(params)
+            # Validating the source requested is the same source
+            for resource in response.json()['data']:
+                actual_source = resource['attributes']['source']
+                self.assertEqual(actual_source, source)
+        # Check more than one source at a time
+        if len(sources) > 0:
+            sample_numbers = random.randint(1, len(sources))
+            rand_sources = random.sample(sources, sample_numbers)
+            sources = ",".join(rand_sources)
+            params = {'studentId': student_id, 'sources': sources}
+            response = self.get_response(params)
+            for resource in response.json()['data']:
+                    actual_source = resource['attributes']['source']
+                    self.assertIn(actual_source, sources)
+
+        # invalid sources tests returns 400
+        invalid_sources = [" ", "advisorportal", "random", "degreeworks"]
+        invalid_source = random.choice(invalid_sources)
+        params = {'studentId': student_id, 'sources': invalid_source}
+        response = utils.make_request(self, endpoint, 400, params=params)
+        schema = utils.get_resource_schema(self, 'Error')
+        utils.check_schema(self, response, schema)
+
+    def query_context_type(self, student_id, context_types, endpoint='/notes'):
+        for context_type in context_types:
+            params = {'studentId': student_id, 'contextTypes': context_type}
+            response = self.get_response(params)
+            # Validating the contextTypes requested
+            # is the same contextTypes received
+            for resource in response.json()['data']:
+                context = resource['attributes']['context']
+                actual_context_type = context['contextType']
+                self.assertEqual(actual_context_type, context_type)
+        # Check more than one contextType at a time
+        sample_numbers = random.randint(0, len(context_types))
+        rand_context_type = random.sample(context_types, sample_numbers)
+        context_types = ",".join(rand_context_type)
+        params = {'studentId': student_id, 'contextTypes': context_types}
+        response = self.get_response(params)
+        for resource in response.json()['data']:
+            context = resource['attributes']['context']
+            actual_context_type = context['contextType']
+            self.assertIn(actual_context_type, context_types)
```

Comment:
Similar comment, you should be able to assert that `actual_context_type` is in `rand_context_type`